### DISCUSSION TITLE
Special treatment to increase accuracy of pressure in lammps  simulation using deepmd potential model?

### DISCUSSION THREAD

**Discussion by hu-lm** [deepmodeling/deepmd-kit#1856] [General]
Hi all,

I am trying to use the potential model trained by deepmd-kit to do extreme high-temperature NVT simulations in lammps (with 2000 bcc Fe atoms). The output from lammps looks good overall except for a significant error in pressure. Thus, I would like to know if there is any specific treatment or choice of, for example, descriptor or anything else that can help to resolve this issue. Thank you so much!
The version I use for deepmd-kit is 2.1.3. Using dpdata to transfer VASP data into deepmd format, with the input below:

{
    "_comment": " model parameters",
    "model": {
	"type_map":	["Fe"],
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		[500],
	    "rcut_smth":	0.50,
	    "rcut":		12.00,
	    "neuron":		[10, 20, 40],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[100, 100, 100],
	    "resnet_dt":	true,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	5000,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"type":		"ener",
	"start_pref_e":	0.02,
	"limit_pref_e":	1,
	"start_pref_f":	1000,
	"limit_pref_f":	1,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "training" : {
	"training_data": {
	    "systems":		["./training_data/"],
	    "batch_size":	"auto",
	    "_comment":		"that's all"
	},
	"validation_data":{
	    "systems":		["./validation_data/"],
	    "batch_size":	"auto",
	    "numb_btch":	1,
	    "_comment":		"that's all"
	},
	"numb_steps":	1000000,
	"seed":		10,
	"disp_file":	"lcurve.out",
	"disp_freq":	1000,
	"save_freq":	10000,
	"_comment":	"that's all"
    },    

    "_comment":		"that's all"
}

**Comment by njzjz** [comment]
You can change `start_pref_v` and `limit_pref_v` to train virials if you have data.

**Reply by hu-lm** [reply]
Hi, thank you so much for your reply. Are there any references or documents that explain these two parameters? Like their meanings and units? For example, if I want to set 382GPa as my start virial, what number should I put in? Thank you!

**Reply by njzjz** [reply]
https://docs.deepmodeling.com/projects/deepmd/en/v2.1.3/train/train-input.html
https://docs.deepmodeling.com/projects/deepmd/en/v2.1.3/data/system.html

**Reply by hu-lm** [reply]
Thank you so much!
Sorry but I still have questions in understanding limit_pref_v(e)(f), so in the document, these were described as: Should be larger than or equal to 0. i.e. the training step goes to infinity.
However, I see that in the water se_e2_a example input script, the limit_pref_e(f) is set to be 1, even for start_pref_e, it was 0.02<1. So I would like to know what is the meaning of 1 here. Shouldn't the limit be set to a much smaller value? And if I can also set the limit_pref_v to 1 as well? Thank you!

**Comment by PabloPiaggi** [comment]
Hi @hu-lm, you said that the pressure has a significant error. How are you measuring that? Are you using the RMS error of the virial? If instead you are seeing large fluctuations of the pressure, it might be normal.

**Reply by hu-lm** [reply]
Hi, thanks for replying.
The significant error in pressure refers to the fact that after the pressure became stable in the lammps simulation, the value of the pressure is still way off the true value. Like 5.6Mpa, and the true value is 4MPa. I am still trying to figure out the virial, but I am pretty sure that I set both start_pref_v and limit_pref_v to 0, which let the deepmd ignore the virial.npy files at first, so that means I wasn't actually using virial at first, and maybe the reason why my pressure is off.
